pkg/objectstore/aws.go

```diff
@@ -40,6 +42,29 @@ func config(region string) *aws.Config {
 	return c
 }
 
+type awsCreds struct {
+	accessKeyID     string
+	secretAccessKey string
+	token           string
+}
+
+// switchRole changes the role using the credentials provider.
+// It returns the new set of credentials.
+func switchRole(awsAccessKeyID, awsSecretAccessKey, role string) (*awsCreds, error) {
+	creds := credentials.NewStaticCredentials(awsAccessKeyID, awsSecretAccessKey, role)
+	sess := session.New(aws.NewConfig().WithCredentials(creds))
+	creds = stscreds.NewCredentials(sess, role)
+	val, err := creds.Get()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get role credentials")
+	}
+	return &awsCreds{
```

Comment:
note: not for this PR, but we should consider using the AWS type `credential.Value`